force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,63 +211,154 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
-        // Loop through each Opportunity name
-            // Create an Opportunity with the given name
+    // Look for account "accountName"; upsert if not there
+    Account account;
+    String stageName = 'Prospecting';
+    Date closeDate = Date.today().addMonths(3);
+
+    List<Account> accounts = [SELECT Id FROM Account WHERE Name = :accountName];
+    
+    if (!accounts.isEmpty()) 
+    {
+        account = accounts[0];
+    } 
+    else 
+    {
+        account = new Account();
+        account.Name = accountName;
+        upsert account;
+    }
+    // Create a list of Opportunities to use for upsert and update
+    List<Opportunity> opportunitiesToUse = new List<Opportunity>();
+
+    // Look for existing Opportunities
+    List<Opportunity> existingOpps = [SELECT Id, Name FROM Opportunity WHERE Name IN :oppNames AND AccountId = :account.Id];
+    
+    // Create a set of existing opportunity names for quick lookup
+    Set<String> existingOppNames = new Set<String>();
+    for (Opportunity opp : existingOpps) {
+        existingOppNames.add(opp.Name);
+    }
+    
+    // Loop through each Opportunity name
+    for (String oppName : oppNames) {
+        if (!existingOppNames.contains(oppName)) {
+            // Create a new Opportunity
+            Opportunity opp = new Opportunity();
+            opp.Name = oppName;
+            opp.StageName = stageName;
+            opp.CloseDate = closeDate;
+            opp.AccountId = account.Id;
```

Comment:
Do we want to do the same thing for existing opps as well? See method instructions:
```
* Create or updates an Opportunities
     * The Opportunity's Stage is set to Qualification.
     * The Opportunity's Close Date is set to 3 months from today.
     * The Opportunity's Amount is set to 50000.
     * @param list of Opportunities to be changed then upserted.
```